applications/tari_app_utilities/src/commands.rs

```diff
@@ -0,0 +1,373 @@
+        _ => Err(CommandError::Argument),
+    }?;
+
+    let inc_amount = match args[3].clone() {
+        Amount(mtari) => Ok(mtari),
+        _ => Err(CommandError::Argument),
+    }?;
+
+    let start_time = match args[4].clone() {
+        Date(dt) => Ok(dt as DateTime<Utc>),
+        _ => Err(CommandError::Argument),
+    }?;
+
+    let pubkey_arg = args[5].clone();
+    let public_key = match pubkey_arg {
+        PublicKey(pk) => Ok(pk),
+        _ => Err(CommandError::Argument),
+    }?;
+
+    let message_arg = args[6].clone();
+    let message = match message_arg {
+        Text(m) => Ok(m),
+        _ => Err(CommandError::Argument),
+    }?;
+
+    // Wait until specified test start time
+    let now = Utc::now();
+    let delay_ms = if start_time > now {
+        println!(
+            "`make-it-rain` scheduled to start at {}: msg \"{}\"",
+            start_time, message
+        );
+        (start_time - now).num_milliseconds() as u64
+    } else {
+        0
+    };
+
+    debug!("make-it-rain delaying for {:?} ms", delay_ms);
+    delay_for(Duration::from_millis(delay_ms)).await;
+
+    let num_txs = (txps * duration as f64) as usize;
+
+    // println!(
+    //     "Making it rain! Sending {} tx/s for {} secs to {}. Starting at {} and incrementing by {}. Message '{}'.",
+    //     txps, duration, public_key, start_amount, inc_amount, message
+    // );
+    let mut tx_ids = Vec::new();
+    // Send all the transactions
+    let start = Utc::now();
+    for i in 0..num_txs {
+        // Manage Tx rate
+        let actual_ms = (Utc::now() - start).num_milliseconds() as u64;
+        let target_ms = (i as f64 / (txps / 1000.0)) as u64;
+        if target_ms - actual_ms > 0 {
+            // Maximum delay between Txs set to 120 s
+            delay_for(Duration::from_millis((target_ms - actual_ms).min(120_000u64))).await;
+        }
+        // Send Tx
+        let amount = start_amount + inc_amount * (i as u64);
+        let send_args = vec![
+            ParsedArgument::Amount(amount),
+            ParsedArgument::PublicKey(public_key.clone()),
+            ParsedArgument::Text(message.clone()),
+        ];
+        let tx_service = wallet_transaction_service.clone();
+        let tx_id = handle
+            .spawn(send_tari(tx_service, send_args))
+            .await
+            .map_err(CommandError::Join)??;
+        debug!("make-it-rain tx_id: {}", tx_id);
+        tx_ids.push(tx_id);
+    }
+
+    Ok(tx_ids)
+}
+
+pub async fn monitor_transactions(
+    transaction_service: TransactionServiceHandle,
+    tx_ids: Vec<u64>,
+    wait_stage: TransactionStage,
+) -> Vec<SentTransaction>
+{
+    let mut event_stream = transaction_service.get_event_stream_fused();
+    let mut results = Vec::new();
+    debug!("monitor transactions wait_stage: {:?}", wait_stage);
+    println!(
+        "Monitoring {} sent transactions to {:?} stage...",
+        tx_ids.len(),
+        wait_stage
+    );
+
+    loop {
```

Comment:
Probably need to work in some sort of timeout and/or way to show the the process is alive and ongoing.

Reply:
There is a timeout one level up where monitor_transactions is called, and the timeout and wait stage are configurable

Reply:
https://github.com/tari-project/tari/pull/2291/files#diff-0aab53ebe00309fc20e74b8823646775R358